mitiq/pt/pt.py

```diff
@@ -72,9 +80,39 @@ def pauli_twirl_circuit(
         twirl_CZ_gates(c, num_circuits=1)[0] for c in CNOT_twirled_circuits
     ]
 
+    if noise_op is not None:
+        twirled_circuits = [
+            add_noise_to_two_qubit_gates(circuit, noise_op)
+            for circuit in twirled_circuits
+        ]
+
     return twirled_circuits
 
 
+@accept_qprogram_and_validate
+def add_noise_to_two_qubit_gates(
```

Comment:
Am I understanding this function correctly, that it adds noise after _all_ CNOT and CZ gates in the circuit?
To solve this issue, we only want noise added to the gates that `pauli_twirl_circuit` adds because we assume the noise operations are already present. 
Apologies if this point was unclear in the original issue.

Reply:
The function defined in this PR could be used before the twirled circuits are returned by the pauli twirling function:

https://github.com/unitaryfund/mitiq/blob/837ea83546f16e39867e3ee74cd27996b1d7509d/mitiq/pt/pt.py#L75

`pauli_twirling_function` would get an optional arg where if a noisy gate is provided, `add_noise_to_two_qubit_gates` is used to return the noisy circuit.

Reply:
Hi @Misty-W  and @purva-thakre , thank you for the comments!
I am also slightly confused with the comment from @Misty-W. As shown in 

https://github.com/EmilianoG-byte/mitiq/blob/efc079fb126353adfa852c196f304ed59794ddaa/mitiq/pt/pt.py#L146

the twirl is applied to all CNOT gates in the circuit, and similarly for the CZ gates. So, as mentioned by @purva-thakre , these are where the noise we want to twirl should be present.

Could you elaborate on:
> we assume the noise operations are already present.

Reply:
I'd thought that line would (unintentionally) cause the noise to be applied twice, but now I see that isn't the case here. Thanks for clarifying, I'll go ahead and resolve this part.